_resources/port1.0/group/python-1.0.tcl

```diff
@@ -447,3 +488,33 @@ pre-livecheck {
         livecheck.type  none
     }
 }
+
+# set-up a test phase
+pre-test {
+    if {[option python.test]} {
+        # return for py-<portname> stub ports, which should not have a test-phase enabled
+        if {[string match py-* [option name]] && ${name} eq ${subport}} {
+            return
+        }
+
+        test.run            yes
+        default test.target {}
+
+        if {[option python.test_framework] eq "pytest"} {
+            default test.args   "-o addopts=''"
+        }
+
+        # set PYTHONPATH depending on whether or not a port is "noarch"
+        if {[option supported_archs] eq "noarch"} {
+            default test.env    "PYTHONPATH=${worksrcpath}/build/lib"
+
+            # if using PEP517, unzip the .whl file to ${worksrcpath}/build/lib
+            if {${python.pep517}} {
+                file mkdir ${worksrcpath}/build/lib
+                system -W ${workpath} "unzip -o *.whl -d ${worksrcpath}/build/lib"
+                }
+        } else {
+            test.env    PYTHONPATH=[glob -nocomplain ${worksrcpath}/build/lib*]
```

Comment:
Doesn't this overwrite any `test.env` that the port may have set? Better append to it.
```suggestion
            test.env-append PYTHONPATH=[glob -nocomplain ${worksrcpath}/build/lib*]
```

Reply:
Or was it your intention to replace the default `test.env` set by the portgroup earlier?

Reply:
I've added the `-append` as that will always be correct. In this case I did want to replace the default `test.env` set earlier. But as Josh pointed out below it is possible to use `PYTHONPATH=[glob -nocomplain ${worksrcpath}/build/lib*]` regardless whether a port is `noarch` or not. It just needs to happen in a `pre-test` phase as the directory doesn't exist the build is finished and thus cannot be done in the "main" Portfile.